### PR TITLE
refactor: make browserCheck an util instead of mixin

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

* Fix https://github.com/nextcloud/spreed/pull/11006#issuecomment-1826876853

- There is no need to be a mixin, it has nothing Vue-related or reactive
- Having this module as a mixin:
  - Limits the usage to only Vue components
  - Reruns checks and all version detections each time the mixin is used (though it is guaranteed that the browser cannot be changed in runtime)
  - Adds many unneeded properties to component instance listed in devtools

### 🖼️ Screenshots / Screencasts

**No visual changes**

### 🚧 Tasks

- [x] Move the module `browserCheck` to utils
- [x] Refactor computed and Vue component methods to exported constants and functions
- [x] Update the usage in components
- [x] Move `callButtonTooltipText` to `CallButton` as it was supposed to be used only in `CallButton`

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖌️ Design was reviewed, approved or inspired by the design team, or there is no visual change
- [ ] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

**Review by Antreesy — APPROVED** [review]
Flawless. Tested in Chrome + Firefox

One non-blocking comment:

File: src/utils/browserCheck.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+const parser = new UAParser()
+const browser = parser.getBrowser()
+
+const getBrowserVersion = () => {
+	if (browser.version) {
+		return browser.version
+	}
+
+	if (browser.name === 'Safari') {
+		// Workaround for https://github.com/faisalman/ua-parser-js/issues/599
```
[the remark this answers is not in the document]
Comment (ShGKme) [reply src/utils/browserCheck.js:38]: Requires Mac for testing =D

[the remark this answers is not in the document]
Comment (nickvergessen) [reply src/utils/browserCheck.js:38]: > Requires Mac for testing =D

No, just set your user agent to 
```
Mozilla/5.0 (iPhone; CPU iPhone OS 15_7 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.6,2 Mobile/15E148 Safari/604.1
```

**Review by SystemKeeper — APPROVED** [review]
Awesome! Also tested with "real safari" works as well :)

**Comment by ShGKme** [comment]
> Awesome! Also tested with "real safari" works as well :)

As I understood the problem, it was not reproducible in Safari on Mac. Only on specific old versions on iPad.

**Pull request merged**